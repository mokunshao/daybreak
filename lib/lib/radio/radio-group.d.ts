import React, { HtmlHTMLAttributes, ChangeEventHandler } from 'react';
import './radio.scss';
export declare const Context: React.Context<{
    val: number;
    e: undefined;
}>;
interface Props extends HtmlHTMLAttributes<HTMLDivElement> {
    value: string;
    onChange: ChangeEventHandler<HTMLInputElement>;
}
export declare const RadioGroup: React.FC<Props>;
export default RadioGroup;
//# sourceMappingURL=radio-group.d.ts.map