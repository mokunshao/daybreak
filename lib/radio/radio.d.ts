import React, { InputHTMLAttributes } from 'react';
import './radio.scss';
export declare const Radio: React.FC<InputHTMLAttributes<HTMLInputElement>>;
export default Radio;
//# sourceMappingURL=radio.d.ts.map