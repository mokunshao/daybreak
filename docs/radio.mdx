---
title: Radio
---

import {useState} from 'react'
import { Playground, Props } from 'docz'
import  Radio from '../src/radio/radio.tsx'
import  RadioGroup from '../src/radio/radio-group.tsx';

# Radio

```javascript
import { Radio, RadioGroup } from 'daybreak';
```

## Basic usage

<Playground>
  {()=>{
    const [value, setValue] = useState('yes');
    const [gender, setGender] = useState('female');
    const onChange = (e) => {
      setValue(e.target.value);
    };
    const onChange2 = (e) => {
      setGender(e.target.value);
    };
    return (
      <div>
        <RadioGroup value={value} onChange={onChange}>
          <Radio name="confirm" value="yes">Yes</Radio>
          <Radio name="confirm" value="no">No</Radio>
        </RadioGroup>
        <br/>
        <RadioGroup value={gender} onChange={onChange2}>
          <Radio name="gender" value="male">Male</Radio>
          <Radio name="gender" value="female">Female</Radio>
        </RadioGroup>
      </div>
    );
  }}
</Playground>
