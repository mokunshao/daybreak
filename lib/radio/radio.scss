.daybreak-radio {
  &-group {
    display: inline-block;
  }

  border: 1px solid red;
}
