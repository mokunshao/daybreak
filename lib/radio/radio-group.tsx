import React, { HtmlHTMLAttributes } from 'react';
import { classes } from '../utils/classes';
import { joinedClass } from '../utils/joinedClass';
import './radio.scss';

const baseClass = joinedClass('radio-group');

interface Props extends HtmlHTMLAttributes<HTMLDivElement> {
  value: string;
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
}

export const RadioGroup: React.FC<Props> = (props) => {
  const { children, className } = props;
  return <div className={classes(baseClass(), className)}>{children}</div>;
};

export default RadioGroup;
