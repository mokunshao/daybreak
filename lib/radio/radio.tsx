import React, { InputHTMLAttributes } from 'react';
import { classes } from '../utils/classes';
import { joinedClass } from '../utils/joinedClass';
import './radio.scss';

const baseClass = joinedClass('radio');

export const Radio: React.FC<InputHTMLAttributes<HTMLInputElement>> = (props) => {
  const {
    children, className, onChange, ...rest
  } = props;
  return (
    <label className={classes(baseClass(), className)}>
      <input className={baseClass('input')} type="radio" {...rest} onChange={onChange} />
      <span className={baseClass('checkmark')} />
      {children}
    </label>
  );
};

export default Radio;
