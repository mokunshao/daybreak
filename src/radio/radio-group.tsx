import React, { HtmlHTMLAttributes, ChangeEventHandler } from 'react';
import { classes } from '../utils/classes';
import { joinedClass } from '../utils/joinedClass';
import './radio.scss';

export const Context = React.createContext({ val: 1212, e: undefined });

const baseClass = joinedClass('radio-group');

interface Props extends HtmlHTMLAttributes<HTMLDivElement> {
  value: string;
  onChange: ChangeEventHandler<HTMLInputElement>;
}

export const RadioGroup: React.FC<Props> = (props) => {
  const {
    children, className, onChange, value,
  } = props;
  return (
    <div className={classes(baseClass(), className)}>
      {React.Children.map(children, (child) => {
        if (!React.isValidElement(child)) { return null; }
        return React.cloneElement(child, { onChange, checked: value === child.props.value });
      })}
    </div>
  );
};

export default RadioGroup;
