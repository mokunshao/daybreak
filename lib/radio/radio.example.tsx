import React, { useState } from 'react';
import { RadioGroup } from './radio-group';
import { Radio } from './radio';

export const RadioExample = () => {
  const [value, setValue] = useState('yes');
  return (
    <>
      <RadioGroup value={value} onChange={(e) => setValue(e.target.value)}>
        <Radio name="confirm" value="yes">Yes</Radio>
        <br />
        <Radio name="confirm" value="no">No</Radio>
      </RadioGroup>
    </>
  );
};

export default RadioExample;
