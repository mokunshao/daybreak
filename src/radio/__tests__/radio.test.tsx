import React, {
  useState, ChangeEventHandler,
} from 'react';
import { render } from '@testing-library/react';
import { Radio } from '../radio';
import { RadioGroup } from '../radio-group';

function Example() {
  const [value, setValue] = useState('yes');
  const onChange: ChangeEventHandler<HTMLInputElement> = (e) => {
    setValue(e.target.value);
  };
  return (
    <div>
      <RadioGroup value={value} onChange={onChange}>
        <Radio name="confirm" value="yes">Yes</Radio>
        <Radio name="confirm" value="no">No</Radio>
      </RadioGroup>
    </div>
  );
}

describe('Radio', () => {
  it('exits', () => {
    expect(Radio).toBeTruthy();
    expect(RadioGroup).toBeTruthy();
    expect(<Example />).toBeTruthy();
  });

  it('match snapshot', () => {
    const { asFragment } = render(<Example />);
    expect(asFragment()).toMatchSnapshot();
  });
});
