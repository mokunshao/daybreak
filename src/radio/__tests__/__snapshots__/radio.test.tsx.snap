// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio match snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="daybreak-radio-group"
    >
      <label
        class="daybreak-radio"
      >
        <input
          checked=""
          class="daybreak-radio-input"
          name="confirm"
          type="radio"
          value="yes"
        />
        <span
          class="daybreak-radio-checkmark"
        />
        Yes
      </label>
      <label
        class="daybreak-radio"
      >
        <input
          class="daybreak-radio-input"
          name="confirm"
          type="radio"
          value="no"
        />
        <span
          class="daybreak-radio-checkmark"
        />
        No
      </label>
    </div>
  </div>
</DocumentFragment>
`;
